lib/hyrax/transactions/steps/set_embargo.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+require 'dry/monads'
+
+module Hyrax
+  module Transactions
+    module Steps
+      ##
+      # Adds and removes work members
+      class SetEmbargo
+        include Dry::Monads[:result]
+
+        ##
+        # @param [Hyrax::Work] obj
+        # @param [Hash] work_members_attributes
+        #
+        # @return [Dry::Monads::Result]
+        def call(work_resource, embargo_set: nil, end_date: nil, visibility_after_embargo: nil, visibility_during_embargo: nil)
+          return Success(work_resource) unless embargo_set.eql?("embargo")
+
+          # I'm not quite sure what to do here.  I think it may be something like this.
+          embargo = Hyrax::Embargo.new(visibility_during_embargo: visibility_during_embargo,
+                                       visibility_after_embargo: visibility_after_embargo,
+                                       embargo_release_date: end_date)
+
+          resource            = Hyrax::Resource.new(embargo: embargo)
+          resource.visibility = 'open'
```

Comment:
shouldn't this be `visibility_during_embargo`?

Reply:
actually, i don't understand what this `resource` is for at all.

Reply:
i think i'm catching up to what you have in mind here.

if you're going to do it this way, you need to set `work_resource.embargo = embargo`. creating a new resource in memory isn't doing anything.

note also that these arguments are existing virtual properties on the ResourceForm changeset. maybe this transaction wants to operate against that changeset to build the embargo before `change_set.apply`?

maybe some unit tests would help guide development?